app/models/user.rb

```diff
@@ -36,6 +40,46 @@ def self.new_with_session(params, session)
     end
   end
 
+  def active_for_authentication?
+    super && check_if_account_active
+  end
+
+  def check_if_account_active
+    return true if Rails.env.test?
+    response = get_user_info
+    if response.status != 200
+      error = JSON.parse(response.body)['error']
+      if error && error == 'invalid_token'
+        refresh_access_token
+        response = get_user_info
+      end
+    end
+    response.status == 200
+  end
+
+  def inactive_message
+    "Sorry, this account has been deactivated."
+  end
+
+  def refresh_access_token
+    conn = Faraday.new(:url => "#{ENV['urs_site']}") do |faraday|
```

Comment:
I am not sure if it needs to be resolved in this PR, but I am wondering if http/api calls should be made in the model (I think more likely it is a no), or if there's a better place to put these http/api calls.

Reply:
Yeah I'd like to as discussed move this logic into the before_action but was having some trouble, so thinking will revisit during IP sprint.